oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -124,18 +124,16 @@ int mca_spml_ucx_del_procs(ompi_proc_t** procs, size_t nprocs)
         mca_spml_ucx_ctx_default.ucp_peers[i].ucp_conn = NULL;
     }
 
-    ret = opal_common_ucx_del_procs(del_procs, nprocs, oshmem_my_proc_id(),
-                                    mca_spml_ucx.num_disconnect,
-                                    mca_spml_ucx_ctx_default.ucp_worker);
-
+    ret = opal_common_ucx_del_procs_nofence(del_procs, nprocs, oshmem_my_proc_id(),
+                                            mca_spml_ucx.num_disconnect,
+                                            mca_spml_ucx_ctx_default.ucp_worker);
+    /* Do not barrier here - barrier is called in _shmem_finalize */
```

Comment:
No need to barrier here - barrier is called in _shmem_finalize

Reply:
fixed